app/plugin_api.go

```diff
@@ -70,6 +71,19 @@ func (api *PluginAPI) UnregisterCommand(teamId, trigger string) error {
 	return nil
 }
 
+func (api *PluginAPI) ExecuteSlashCommand(commandArgs *model.CommandArgs) (*model.CommandResponse, error) {
```

Comment:
I'm a bit worried that this naming collides  with the Hook  `ExecuteCommand`. Maybe something like `TriggerSlashCommand`?

Reply:
I can see both arguments. At least `ExecuteCommand` is a hook, so that might be enough to remove ambiguity?

Reply:
Ya, not totally happy ether. I originally had it called `ExecuteCommand` then had the obvious conflict. I like ` ExecuteSlashCommand` over `TriggerSlashCommand` because it matches what we call that server side. Since it doesn't directly conflict with the hook I think it's fine.

Reply:
What I like about `TriggerSlashCommand` is that it makes clear that the caller only _triggers_ a slash command. Other then with `ExecuteCommand`, where the plugin has to provide the implementation.

For a new developers trying out the Plugin toolkit it might not be obvious what the difference between a hook and an API call is here.

Reply:
As far as I am concerned `Trigger` and `Execute` in this context are synonyms. I agree that this might be confusing, but I don't see how to handle that other than docs. I would rather be internally consistent.

Reply:
To me `Execute` means "Do something" while `Trigger` means "Make someone do something".

What is the reasoning for valuing internal consistency vs external consistency? Given that this is an API endpoint I would rate a clear and easy to use API higher then internal consistency.

Can we rename our internal code to align with the external consistency?

Reply:
As @cpoile noted, I think hooks and APIs are sufficiently distinct so as to not worry about a conflict over the name `ExecuteCommand`, especially since the author must call one and implement a receiver function for the other.

I also note the REST API explicitly [calls this endpoint](https://api.mattermost.com/#tag/commands/paths/~1commands~1execute/post) `Execute Command`. I don't see anywhere where `trigger` is used as a verb: it's exclusively referring to the `trigger word` in a slash command, so I'd be hesitant to adopt new terminology for this particular change.

Can we just keep this as `ExecuteCommand`, and potentially iterate on improved semantics in the plugin-api wrapper in the future?